src/core/createEventManager.js

```diff
@@ -74,7 +78,8 @@ export default ({
           // if the callback returns false, the event should not be sent
           if (!event.shouldSend()) {
             onRequestFailureCallbackAggregator.add(lifecycle.onRequestFailure);
-            const error = new Error("Event was cancelled.");
+            logger.info(EVENT_CANCELLATION_MESSAGE);
```

Comment:
This is the meat of the PR where we actually log a message.

As you can see, I made the message say `Event was canceled because the onBeforeEventSend callback returned false.` rather than just `Event was canceled.` because I think that gives the customer a better idea of how the event was canceled.

With this in mind, I was on the fence as to whether we should log this message here in `createEventManager.js` or `createEvent.js` (where we actually call `onBeforeEventSendCallback`). From the perspective of the event manager, the event manager only knows that the event shouldn't be sent. It doesn't necessarily know for certain that the reason it shouldn't be sent is that `onBeforeEventSend` returned `false`. From the perspective of the event, the event knows that the `onBeforeEventSend` callback returned `false` and therefore the event makes its `shouldSend` method return `false`, but it technically doesn't know that the event will not actually be sent or that it's not going to be sent due to other potential reasons as well.

Anyway, I'm curious if you have opinions here.

Reply:
You could add a method on the event `getCancelReason` or something similar. That way the event could keep track of why the event shouldn't be sent and then the eventManager can decide what to do with that information. For now, though, there is only one possible reason that the event shouldn't be sent namely that false was returned. When an error is thrown, there is a different reason, but then the error is thrown from finalize. So maybe it doesn't make sense to add this method.

Reply:
Yeah, that's similar to what I was thinking as well. When the time comes that are more reasons for the event to not be sent, we can play with event's API.